test/integration_test.exs

```diff
@@ -179,4 +195,12 @@ defmodule ConfigCat.IntegrationTest do
 
     assert %ArgumentError{message: "SDK Key is required"} = error
   end
+
+  defp clear_registry(_context) do
+    ConfigCat.Registry
+    |> Registry.match({__MODULE__, :_}, :_)
+    |> Enum.each(fn {key, _} -> Registry.unregister(ConfigCat.Registry, key) end)
+
+    :ok
+  end
```

Comment:
Is this part strictly necessary? Process should remove themselves from the Registry when they shut down, and the use of `start_supervised!` ensures that processes are shut down when the test ends.

If it is necessary, are you sure that this code is doing what you want? `__MODULE__` in the match spec will refer to this test's module (`ConfigCat.IntegrationTest`), but all of the started processes will have the module as `ConfigCat.Supervisor`, if I'm remembering correctly.

Reply:
My first attempt was to simply turn off async test execution in ConfigCatTest. But on GitHub I still got the error. After that, I implemented the `clear_registry` function to ensure that the previous ConfigCat client instance is removed from the Registry. If I remember correctly, even with these changes, I occasionally encountered the error (though less frequently). Finally, I combined the two separate tests, and the error disappeared.

Reply:
I'm pretty sure that `clear_registry` isn't doing anything (because `__MODULE__` is different in the test than it is in the code), but if this is working for you, feel free to merge it.

Reply:
From what I can see, we're already using `start_supervised` in `start_config_cat`, which is what every test calls. Clearly something isn't shutting down fast enough, though. I have a different idea to try. I'll open up a draft PR once I have it working.

Reply:
@kp-cat See #143 for an alternate solution for this test problem.

If we find that the tests continue spuriously failing, we could change our CI config to run `mix coveralls.json --warnings-as-errors --max-cases 1`. That would force the tests to all run serially. That'll make the build a bit slower, but should make the problem go away.